bittensor/_axon/__init__.py

```diff
@@ -412,8 +412,12 @@ def parse_signature(
     ) -> Tuple[int, str, str, str, int]:
         r"""Attempts to parse a signature from the metadata"""
         signature = metadata.get("bittensor-signature")
+        version = metadata.get('bittensor-version')
         if signature is None:
             raise Exception("Request signature missing")
+        if version < 370:
+            raise Exception("Incorrect Version")
+        
         for parser in [self.parse_signature_v2, self.parse_legacy_signature]:
```

Comment:
You can just drop the `self.parse_legacy_signature` from below. Clients older than 3.6.0 will send such signatures and will be rejected.